### PR TITLE
LP-1744 Enroll all courses of a specialisation on single click 

### PR DESCRIPTION
**Pull request by zahid-arbisoft**
### Description

[LP-1744](https://philanthropyu.atlassian.net/browse/LP-1744)

[philu-edx-theme PR](https://github.com/philanthropy-u/philu-edx-theme/pull/1444)

Notes:

1. Code include all test cases
2. Mocking package `pytest-mock==2.0.0` added to philu base requirement file. I have verified that it does not change version of any other package

### CODE REVIEW

**Review by qasimgulzar — APPROVED** [review]
LGTM

File: openedx/features/specializations/helpers.py [hunk 1]
```diff
@@ -1,7 +1,147 @@
...
+        sorted(
+            open_course_rerun_list, key=lambda open_rerun: date_from_str(open_rerun['enrollment_start'])
+        )[0] if opened else course_runs[0]
+    )
+
+    course_rerun['opened'] = opened
+    return course_rerun
+
+
+def get_first_chapter_link(course_id):
```
Comment (muak50) [inline openedx/features/specializations/helpers.py:97]: We have a function to get link of any chapter of provided course... Pls have a look at `from openedx.features.philu_courseware.helpers import get_nth_chapter_link` if that could server our purpose and we don't have to write an additional function here.

**Review by muak50 — APPROVED** [review]
Kindly have a look at couple of suggestion and reply them... Overall LGTM 👍 

Great work!

File: cms/envs/test.py [hunk 2]
```diff
@@ -371,3 +371,8 @@
 ########################## Derive Any Derived Settings  #######################
 
 derive_settings(__name__)
+
+
+DEFAULT_SITE_THEME = 'philu'
+COMPREHENSIVE_THEME_DIRS = ['/edx/src/philu-edx-theme/edx-platform/']
+ENABLE_COMPREHENSIVE_THEMING = True
```
[the remark this answers is not in the document]
Comment (zahid-arbisoft) [reply cms/envs/test.py]: Done

File: openedx/features/specializations/helpers.py (hunk 1 above)
Comment (zahid-arbisoft) [reply openedx/features/specializations/helpers.py:97]: As per our discussion leaving this as is.

**Pull request merged**